code.py

```diff
@@ -4,23 +4,50 @@
 It enables you to more easily create complex layouts and has other advanced 
 features, including double-tap support.
 """
+try:
+    from apps_private.default import DEFAULT_APP, SELECT_APPS
+except Exception:
+    from apps.home import HomeApp
+
+    DEFAULT_APP = HomeApp
+    SELECT_APPS = {}
 
-from apps.home import HomeApp
 from utils.app_pad import AppPad
 from utils.commands import AppSwitchException
+from utils.config import conf
 from utils.constants import OS_MAC, OS_SETTING, PREVIOUS_APP_SETTING
 
-app_settings = {
-    OS_SETTING: OS_MAC,
-    PREVIOUS_APP_SETTING: [],
-}
 
-app_pad = AppPad()
-current_app = HomeApp(app_pad, app_settings)
+class SettingsAppPad(AppPad):
+    def main_loop_hook(self):
+        if conf.reload_config():
+            raise AppSwitchException(None)
+
+
+APP_PAD = SettingsAppPad()
+
+
+def get_settings():
+    app_settings = {OS_SETTING: OS_MAC, PREVIOUS_APP_SETTING: []}
+
+    for k, v in conf.__dict__.items():
+        app_settings[k] = v
+
+    return app_settings
+
+
+def get_app():
+    if process := get_settings().get("host", {}).get("process"):
+        print(process.lower(), SELECT_APPS.get(process.lower(), DEFAULT_APP))
+        return SELECT_APPS.get(process.lower(), DEFAULT_APP)
+    return DEFAULT_APP
+
+current_app = get_app()(APP_PAD, get_settings())
 
 while True:
     try:
         print(f"Current App = {current_app}")
         current_app.run()
     except AppSwitchException as err:
-        current_app = err.app
+        current_app = err.app if err.app else get_app()(APP_PAD, get_settings())
```

Comment:
When changing the app this way, the `SwitchAppCommand` is not used which leads to `PreviousAppCommand` throwing an `IndexError`. 

Perhaps the `AppSwitchException` should use `SwitchAppCommand`?